src/main/java/seedu/eduke8/ui/Ui.java

```diff
@@ -351,33 +381,41 @@ public void addNoteInteractions(TopicList topicList) {
 
     public void deleteNoteInteractions(TopicList topicList) {
         Ui ui = new Ui();
+        int noteCount = 0;
+        Topic topic = null;
+        NoteList noteList = null;
 
         System.out.println(HORIZONTAL_LINE);
         System.out.println(DELETE_NOTE_PROMPT_FOR_TOPIC);
-        String topicName = SCANNER.nextLine().trim();
 
+        String topicName = SCANNER.nextLine();
+        
         try {
-            if (topicList.doesTopicExist(topicName)) {
-                Topic topic = (Topic) topicList.find(topicName);
-                NoteList noteList = topic.getNoteList();
-                ui.printNoteList(noteList);
-
-                System.out.println(DELETE_NOTE_PROMPT_FOR_INDEX);
-                String input = SCANNER.nextLine().trim();
-
-                if (input.matches("[0-9]+") && Integer.parseInt(input) > 0
-                        && Integer.parseInt(input) <= noteList.getCount()) {
-                    int index = Integer.parseInt(input);
-                    topic.getNoteList().delete(index - 1);
-                    printMessage(DELETE_NOTE_SUCCESSFULLY);
-                } else {
-                    printMessage(INVALID_TOPIC_INDEX + System.lineSeparator() + DELETE_NOTE_UNSUCCESSFULLY);
-                }
+            topic = (Topic) topicList.find(topicName);
+            noteList = topic.getNoteList();
+            noteCount = noteList.getCount();
+        } catch (Eduke8Exception e) {
+            ui.printError(e.getMessage());
+        }
+
+        if (topicList.doesTopicExist(topicName) && noteCount > 0) {
+            ui.printNoteList(noteList);
+
+            System.out.println(DELETE_NOTE_PROMPT_FOR_INDEX);
+            String input = SCANNER.nextLine();
+
+            if (input.matches("[0-9]+") && Integer.parseInt(input) > 0
```

Comment:
Take note the magic literal here, might want to replace it with a String constant!